juju/machine.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, *args, **kwargs):
                                            loop=model.loop)
         if model._watch_stopping.is_set():
             return
-        if model.info.agent_version < client.Number.from_json('2.2.3'):
+        if 'agent_version' in model.info and model.info.agent_version < client.Number.from_json('2.2.3'):
```

Comment:
in newer versions of `ModelInfo` the `agent_version` is gone, so we need to check that it exists before checking.

Reply:
Further investigation, this wasn't the case, we just wasn't exporting the right ModelInfo. I've fixed this so that we do indeed do that correctly.